src/main/kotlin/io/zerobase/smarttracing/resources/Amazon.kt

```diff
@@ -0,0 +1,179 @@
+import javax.mail.internet.InternetAddress
+import javax.mail.internet.MimeMessage
+import javax.mail.internet.MimeMultipart
+import javax.mail.internet.MimeBodyPart
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.util.Properties
+import software.amazon.awssdk.core.SdkBytes
+import software.amazon.awssdk.services.ses.model.SendRawEmailRequest
+import software.amazon.awssdk.services.ses.model.RawMessage
+import io.zerobase.smarttracing.AmazonConfig
+
+// Currently only implements the AmazonSES
+class Amazon(private val config: AmazonConfig) {
+    private val session: Session
+    private val client: SesClient
+
+    init {
+        session = Session.getDefaultInstance(Properties())
+        client = SesClient.builder().region(Region.US_EAST_1).build()
+    }
+
+    /**
+     * Sends a normal email (abstraction to simplify sending messages)
+     *
+     * @param subject subject to send this under
+     * @param html html to send
+     * @param to address to send to
+     *
+     * @throw AddressException when the address is incorrect
+     * @throw MessagingException when we cannot message
+     * @throw IOException input output exception
+     */
+    private fun sendNormal(subject: String, html_string: String, to: String) {
+        if (config.from != "") {
+            val message = MimeMessage(session)
+
+            message.setSubject(subject, "UTF-8")
+            message.setFrom(InternetAddress(config.from))
+            message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(to))
+
+            val msgBody = MimeMultipart("alternative")
+            val wrap = MimeBodyPart()
+            val html = MimeBodyPart()
+
+            html.setContent(html_string, "text/html; charset=UTF-8")
+            msgBody.addBodyPart(html)
+            wrap.setContent(msgBody)
+
+            val msg = MimeMultipart("mixed")
+
+            message.setContent(msg)
+            msg.addBodyPart(wrap)
+
+            val outStream = ByteArrayOutputStream()
+
+            message.writeTo(outStream)
+
+            val arr = outStream.toByteArray()
+
+            val data = SdkBytes.fromByteArray(arr)
+            val rawMessage = RawMessage.builder().data(data).build()
+            val rawEmailRequest = SendRawEmailRequest.builder().rawMessage(rawMessage).build()
+
+            client.sendRawEmail(rawEmailRequest)
+        }
+    }
+
+    /**
+     * Sends a qr code email (abstraction to simplify sending messages)
+     *
+     * @param subject subject to send this under
+     * @param html html to send
+     * @param to address to send to
+     * @param url the url to encode in a qr code and send
+     *
+     * @throw AddressException when the address is incorrect
+     * @throw MessagingException when we cannot message
+     * @throw IOException input output exception
+     */
+    private fun sendQRScannable(subject: String, html_string: String, to: String, url: String) {
```

Comment:
the email abstraction shouldn't know anything about qr codes

Reply:
it needs to be able to handle qr codes as an attachment, as such i thought it would make more sense for this section to generate the pdf here as opposed to passing it from function to function

Reply:
no it needs to be able to send attachments. QR codes are an attachment. the email abstraction should *not* know about what the attachments are. it should take at most an address, a template reference to render, the model to render, and any attachments.

Reply:
alright i will set that up